bank_account.rb

```diff
@@ -0,0 +1,92 @@
+require 'date'
+
+class BankAccount
+  attr_accessor :name, :job, :email, :address, :pin, :balance, :last_withdrawal_date, :remaining_daily_withdrawal_limit
+  
+  WITHDRAWAL_LIMIT_PER_DAY = 5000
+
+  def initialize(name, job, email, address)
+    @name = name
+    @job = job
+    @email = email
+    @address = address
+    @pin = nil
+    @balance = 0.0
+    @last_withdrawal_date = nil
+    @remaining_daily_withdrawal_limit = WITHDRAWAL_LIMIT_PER_DAY
+  end
+
+  def set_pin(pin)
+    @pin = pin
+  end
+
+  def deposit_amount(amount)
+    if @pin.nil?
+      puts "PIN is required before insertin deposit."
```

Comment:
You should not print things in a class that handles logic, that should be done in the BankSystem